Mathlib/LinearAlgebra/Dimension/FreeAndStrongRankCondition.lean

```diff
@@ -244,3 +244,77 @@ theorem cardinal_lt_aleph0_of_finiteDimensional (K V : Type u) [Ring K] [StrongR
   rw [cardinal_mk_eq_cardinal_mk_field_pow_rank K V]
   exact Cardinal.power_lt_aleph0 (Cardinal.lt_aleph0_of_finite K) (rank_lt_aleph0 K V)
 #align cardinal_lt_aleph_0_of_finite_dimensional cardinal_lt_aleph0_of_finiteDimensional
+
+theorem Subalgebra.eq_bot_of_rank_le_one
```

Comment:
Can you declare `F` and `E` only once? Or maybe using the variables already in scope.